src/App.js

```diff
@@ -14,11 +15,91 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  selectGood = good => this.setState(prevState => ({
+    selectedGoods: [...prevState.selectedGoods, good],
+  }));
+
+  deleteGood = good => (
+    this.setState((prevState) => {
+      const allGoods = [...prevState.selectedGoods];
+
+      allGoods.splice(allGoods.indexOf(good), 1);
+
+      return {
+        selectedGoods: [...allGoods],
+      };
+    })
+  );
+
+  clearGoods = () => this.setState({ selectedGoods: [] });
+
+  title = (goods) => {
+    if (goods.length === 0) {
+      return <span>No goods selected</span>;
+    }
+
+    if (goods.length === 1) {
+      return <span>{`${goods[0]} is selected`}</span>;
+    }
+
+    const result = goods.slice(0, goods.length - 1).join(', ');
+
+    return <span>{`${result} and ${goods[goods.length - 1]}`}</span>;
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>
+          {this.title(this.state.selectedGoods)}
+          {this.state.selectedGoods.length !== 0 && (
+            <button
+              onClick={this.clearGoods}
+              type="button"
+            >
+              X
+            </button>
+          )}
+        </h1>
+        <span>{goodsFromServer.length}</span>
+        <ul>
+          {goodsFromServer.map(good => (
+            <li
+              key={good}
+              className={classNames({
+                selected: this.state.selectedGoods.includes(good),
```

Comment:
you can create some variable (`isIncluded` for example) inside callback in `map` method before you return `<li>`, and use this variable in all places where you need it. with this approach, you can use the method includes only once